contracts/Ida.sol

```diff
@@ -84,22 +88,40 @@ contract Ida {
         emit Funded(msg.sender, _amount, impactPromise.totalSupply());
     }
 
+    function registerReport(bytes32 key) public onlyValidator {
+      require(!hasEnded(), "Cannot register after project end");
+      require(!reportRegistered[key], "The report is already registered!");
+      reportRegistered[key] = true;
+      require(claimsRegistry.getClaim(serviceProvider, address(this), key) == bytes32(promisePrice), "A claim must be registered before registering a report");
```

Comment:
Shouldn't we also check:
require(!claimsRegistry.isApproved(validator, serviceProvider, address(this), key), "This promise has already been validated");
to avoid reporting for already validated promises ?

Reply:
I didn't put it there because on validator's side we only allow to register new reports. `require(report.status == Status.None, "The report for this impact promise has already been created.");`
So if the report has been created and even approved, it wouldn't be possible to translate it to Ida.
But I agree that this check still can be added for readability and consistency, and also for extra security.